src/peer_manager.rs

```diff
@@ -391,9 +392,32 @@ impl PeerManager {
         (ids_to_drop, our_new_prefix)
     }
 
+    /// Checks whether we have a quorum of nodes in each section
+    fn is_merging_possible(&self) -> bool {
+        let prefixes = self.expected_peers
+            .keys()
+            .map(|x| self.routing_table.find_group_prefix(x))
+            .collect::<HashSet<_>>();
+        if prefixes.contains(&None) {
+            // we expect contacts that don't belong in any of the sections in our RT - so we have
+            // no contacts from their section
+            return false;
+        }
+        // `unwrap` is safe here - we just verified that there is no `None` in the set
+        for prefix in prefixes.into_iter().map(Option::unwrap) {
```

Comment:
It's still avoidable, and to make the actual `unwrap`s that could potentially be problematic easier to find, let's avoid them where possible. E.g. you could just `flat_map(|x| x)` instead.